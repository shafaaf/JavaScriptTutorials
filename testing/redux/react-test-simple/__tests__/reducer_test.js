import reducer, { initialState} from '../src/redux/reducer';

describe('todos reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle TODO_REQUEST', () => {
    expect(
      reducer(initialState,
        {
          type: 'TODO_REQUEST',
        }),
    ).toMatchSnapshot();
  });
});
