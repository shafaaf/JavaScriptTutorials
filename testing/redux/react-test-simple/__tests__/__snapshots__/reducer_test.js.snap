// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos reducer should handle TODO_REQUEST 1`] = `
Object {
  "error": null,
  "requesting": true,
  "todos": Array [],
}
`;

exports[`todos reducer should return the initial state 1`] = `
Object {
  "error": null,
  "requesting": false,
  "todos": Array [],
}
`;
